version: '2.1'

volumes:
  django_sample_db_dev: {}
  django_sample_packages: {}

services:
  db:
    image: postgres:9.6.1
    volumes:
      - /home/ec2-user/postgresql-greatchoice:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=greatchoice
      - POSTGRES_USER=greatchoice
      - POSTGRES_PASSWORD=chorock12#
      - POSTGRES_INITDB_ARGS=--encoding=UTF-8
    ports:
      - "5432:5432"
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -U postgres"
      interval: 3s
      timeout: 1s
      retries: 10

  django:
    build:
      context: .
      dockerfile: ./compose/django/Dockerfile-dev
    environment:
      - DJANGO_DEBUG=True
      - DJANGO_DB_HOST=postgres
      - DJANGO_DB_PORT=5432
      - DJANGO_DB_NAME=greatchoice
      - DJANGO_DB_USERNAME=greatchoice
      - DJANGO_DB_PASSWORD=chorock12#
      - DJANGO_SECRET_KEY=dev_secret_key
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    links:
      - db:postgres
    command: /start-dev.sh
    volumes:
      - ./manage.py:/app/manage.py
      - ./requirements.txt:/app/requirements.txt
      - ./djangosample:/app/djangosample
      - django_sample_packages:/usr/local/lib/python3.6/site-packages/
